segm8/segm8.py

```diff
@@ -0,0 +1,358 @@
+        split_number = list(format(number, format_spec))
+        placeholder = "0" if pad_zeros and align == Align.RIGHT else " "
+        self._display(split_number, position, width, align, placeholder)
+
+    def display_string(self, string, position, width, align=Align.LEFT):
+        """Prints a string of a fixed width starting with the position
+        to the output buffer using formatting flags. Moves the contents
+        of the buffer to the indicator daisy-chain. Note: the "." sign
+        does not occupy a  separate module and is displayed with the
+        previous character.
+
+        Parameters:
+        -----------
+        string: str
+            Text line.
+        position: int
+            The starting position of the output of a string in the
+            internal buffer.
+        width: int
+            The number of elements in the internal buffer, and
+            accordingly the number of SegM8 modules needed to display
+            a number or a text.
+
+        Optional parameters:
+        --------------------
+        align: int
+            Output alignment method. "segm8.Align.RIGHT" – align to the
+            right corner, "segm8.Align.LEFT" – align to the left corner.
+
+        Returns:
+        --------
+        None
+        """
+        split_string = list(string)
+        self._display(split_string, position, width, align)
+
+    def write_segments(self, mask, device_index):
+        """Displays a custom symbol in the specified position.
+
+        Parameters:
+        -----------
+        mask: int
+            list of boolean values for all 8 segments. The ordinal
+            number of the member of this list corresponds to the letter
+            index of a indicator segment: 0-a, 1-b, <...>, 7-h(dot).
+        device_index: int
+            A SegM8 device ordinal number in the daisy-chain. Can be in
+            the range [0 .. device_count - 1].
+
+        Returns:
+        --------
+        None
+        """
+        for i, val in enumerate(mask):
+            if val:
+                self._data[device_index] |= (1 << i) & 0xFF
+            else:
+                self._data[device_index] &= ~(1 << i) & 0xFF
+
+        self._send_data()
+
+    def __del__(self):
+        """The destructor is used to set all segments of all modules to
+        the "dark" state and to close the SPI connection."""
+        self._data = [font.FONT[" "]] * self._device_count
+        self._spi.writebytes(self._data)
+        self._spi.close()
+
+    def _spi_init(self, bus, pin_CE):
+        """Returns a new SPI object that is connected to the specified
+        SPI device interface.
+
+        Parameters:
+        -----------
+        bus: int
+            SPI bus number.
+        pin_CE: int
+            The chip enable pin.
+
+        Returns:
+        --------
+        SpiDev object
+        """
+        spi = spidev.SpiDev()
+        spi.open(bus, pin_CE)
+        spi.max_speed_hz = 62500000
```

Comment:
"spi.max_speed_hz = 62500000"
I'm not sure if these modules will be able to work stably at this SPI frequency. Especially if the chain is long. I was only able to test them up to 1 MHz

Reply:
I was unable to properly display data at frequencies lower than this on the RPi.